ops/src/slack.rs

```diff
@@ -243,11 +256,55 @@ impl Slack {
 
         Ok(resp.user)
     }
+
+    pub async fn lookup_previous_post(&self, thread_url: String, slack_channel: String) -> Result<Option<String>> {
+        #[derive(Deserialize)]
+        struct SearchMatch {
+            ts: Option<String>,
+        }
+        #[derive(Deserialize)]
+        struct Messages {
+            matches: Vec<SearchMatch>,
+            total: i32,
+        }
+        #[derive(Deserialize)]
+        struct Resp {
+            ok: bool,
+            error: Option<String>,
+            messages: Option<Messages>,
+        }
+
+        let resp: Resp = self
+            .client
+            .get(&format!("/search.messages?query='in:{} from:@octobot has:link {}'&sort=timestamp", slack_channel, thread_url))
```

Comment:
hmm. i fear this is a little fragile and also too expensive. This call is "tier 2" rate limited -- only allowed 20 per minute. 

i think we'll need to keep track ourselves of which message / thread ids we've created that belong to each PR. (and then prune that list over time too)

i also think we may not want to use the `html_url` of the PR as the identifier at that point. Maybe want to use a composite key like `(owner, repo, pr_num)`.